In [3]:
#install packages
install.packages("seminr")


The downloaded binary packages are in
	/var/folders/_8/_r3pgsjx2l74qstm6wlz1m8r0000gr/T//RtmprxKRWV/downloaded_packages


In [4]:
#load packages
library(seminr)

In [10]:
# Load the corporate reputation data
corp_rep_data<-read.csv(file = 'Corporate Reputation Data.csv', header = TRUE, sep = ';')

In [11]:
# Show the first several rows of the corporate reputation data
head(corp_rep_data)

serviceprovider servicetype comp_1 comp_2 comp_3 like_1 like_2 like_3 cusl_1
1 3               2           4      5      5      3      1      2      5     
2 3               2           6      7      6      6      6      6      7     
3 3               2           4      5      2      5      5      5      7     
4 3               2           6      4      4      6      5      6      7     
5 3               2           6      4      6      6      6      7      6     
6 3               2           3      4      4      6      7      7      7     
  cusl_2 <U+22EF>        qual_4 qual_5 qual_6 qual_7 qual_8 qual_global switch_1
1 3      <U+22EF> 2      4      4      2      5      6           3       
2 7      <U+22EF> 6      5      6      6      5      6           5       
3 7      <U+22EF> 4      3      5      4      4      5           4       
4 7      <U+22EF> 7      5      4      5      6      7           3       
5 7      <U+22EF> 7      6      6      6      7      7           5       
6 7      <U+22EF> 6      5      6      3      4      6           5       
  switch_2 switch_3 switch_4
1 1        3        2       
2 5        4        4       
3 3        2        3       
4 4        4        2       
5 5        5        4       
6 2        4        4

In [13]:
# Create measurement model 
simple_mm <- constructs(
    composite('COMP', multi_items('comp_', 1:3)), 
    composite('LIKE', multi_items('like_', 1:3)), 
    composite('CUSA', single_item('cusa')), 
    composite('CUSL', multi_items('cusl_', 1:3)))

In [14]:
# Create structural model 
simple_sm <- relationships(
    paths(from = c('COMP', 'LIKE'), to = c('CUSA', 'CUSL')), 
    paths(from = c('CUSA'), to = c('CUSL')))

In [16]:
# Estimate the model
corp_rep_simple_model <- estimate_pls(
    data = corp_rep_data, 
    measurement_model = simple_mm, 
    structural_model = simple_sm, 
    inner_weights = path_weighting, 
    missing = mean_replacement, 
    missing_value = '-99')

Generating the seminr model

All 344 observations are valid.



In [18]:
# Estimate the model with default settings
corp_rep_simple_model <- estimate_pls(
    data = corp_rep_data, 
    measurement_model = simple_mm,
    structural_model = simple_sm,
    missing_value = '-99')

Generating the seminr model

All 344 observations are valid.



In [19]:
# Summarize the model results
summary_simple_corp_rep <- summary(corp_rep_simple_model)

In [20]:
# Inspect the model’s path coefficients and the R^2 values
summary_simple_corp_rep$paths

CUSA      CUSL       
R^2    0.2945679 0.562038760
AdjR^2 0.2904304 0.558174396
COMP   0.1620512 0.009174543
LIKE   0.4238696 0.342069437
CUSA          NA 0.504466308

In [21]:
# Inspect the construct reliability metrics 
summary_simple_corp_rep$reliability

alpha     rhoC      AVE       rhoA     
COMP 0.7760283 0.8646151 0.6805816 0.8318793
LIKE 0.8309866 0.8985664 0.7470805 0.8360573
CUSA 1.0000000 1.0000000 1.0000000 1.0000000
CUSL 0.8309882 0.8990671 0.7484172 0.8390415

In [23]:
# Bootstrap the model
boot_simple_corp_rep <- bootstrap_model(
    seminr_model = corp_rep_simple_model,
    nboot = 1000, 
    cores = NULL, 
    seed = 123)

Bootstrapping model using seminr...

SEMinR Model successfully bootstrapped



In [24]:
# Store the summary of the bootstrapped model
sum_boot_simple_corp_rep <- summary(boot_simple_corp_rep)

In [25]:
# Inspect the bootstrapped structural paths 
sum_boot_simple_corp_rep$bootstrapped_paths

Original Est. Bootstrap Mean Bootstrap SD T Stat.   2.5% CI    
COMP  ->  CUSA 0.162051225   0.16606946     0.06825057    2.374357  0.03765878
COMP  ->  CUSL 0.009174543   0.01100749     0.05565154    0.164857 -0.09778233
LIKE  ->  CUSA 0.423869606   0.42236783     0.06180983    6.857641  0.29945525
LIKE  ->  CUSL 0.342069437   0.34042290     0.05646014    6.058601  0.22686838
CUSA  ->  CUSL 0.504466308   0.50419317     0.04211729   11.977654  0.41871043
               97.5% CI 
COMP  ->  CUSA 0.2984737
COMP  ->  CUSL 0.1260371
LIKE  ->  CUSA 0.5420963
LIKE  ->  CUSL 0.4498754
CUSA  ->  CUSL 0.5850431

In [26]:
# Inspect the bootstrapped indicator loadings 
sum_boot_simple_corp_rep$bootstrapped_loadings

Original Est. Bootstrap Mean Bootstrap SD T Stat.  2.5% CI  
comp_1  ->  COMP 0.8577188     0.8576521      2.073469e-02 41.36636 0.8125353
comp_2  ->  COMP 0.7984526     0.7968225      2.881530e-02 27.70933 0.7300102
comp_3  ->  COMP 0.8176411     0.8140940      3.115288e-02 26.24609 0.7462230
like_1  ->  LIKE 0.8793263     0.8796771      1.721186e-02 51.08839 0.8425527
like_2  ->  LIKE 0.8702393     0.8685666      1.834869e-02 47.42787 0.8300191
like_3  ->  LIKE 0.8430364     0.8418949      2.035482e-02 41.41704 0.7985460
cusa  ->  CUSA   1.0000000     1.0000000      2.798047e-15       NA 1.0000000
cusl_1  ->  CUSL 0.8328272     0.8316624      2.357204e-02 35.33115 0.7802457
cusl_2  ->  CUSL 0.9172732     0.9170540      1.032111e-02 88.87353 0.8941109
cusl_3  ->  CUSL 0.8427695     0.8417575      2.269544e-02 37.13388 0.7928199
                 97.5% CI 
comp_1  ->  COMP 0.8950192
comp_2  ->  COMP 0.8428669
comp_3  ->  COMP 0.8655297
like_1  ->  LIKE 0.9096542
like_2  ->  LIKE 0.9002398
like_3  ->  LIKE 0.8787622
cusa  ->  CUSA   1.0000000
cusl_1  ->  CUSL 0.8736887
cusl_2  ->  CUSL 0.9354416
cusl_3  ->  CUSL 0.8810509

In [28]:
# Write the bootstrapped paths object to csv file
write.csv(
    x = sum_boot_simple_corp_rep$bootstrapped_loadings, 
    file = 'boot_loadings.csv')

<h1>How to do a PLS-SEM write up</h1>

- PLS-SEM write up 
	- Determine rather a construct is formative or reflective  
		- Determined using theoretical constructs  
			construct is reflective when items forming a latent constructs are interchangeable and theoretically points to a single dimension, while formative constructs are theoretically distinct and thus the items are not interchangeable {Diamantopoulos, 2001 #1185}

		- Determined convergent and discriminant validity of constructs  
			Since model fits are covariance based, they are less applicable to PLS model estimations {Hair, 2019 #1439}. Instead, factorial validity is estavblished through finding statistical support for convergent and discriminant validities {Straub, 2004 #1912}, which are assessed through comparing the respective indicators to their respective thresholds.

	- Establishing factorial validities of Reflective Constructs  
		- Indicator loadings  
			item loadings for a latent variable is significant {Moody, 2014 #1022}, with a loading of .708 or higher {Hair, 2019 #1439}

		- internal consistency reliability  
			Cronbach’s alpha and composite reliability that is higher than .70 (or .60) and below .90 is considered acceptable {Hair, 2019 #1439}, 

		- Convergent validity  
			Convergent validity, which refers to “the extent to which the converges to explain the variance of its items,” {Hair, 2019 #1439@9} is evaluated by average variance extracted (AVE) which should be larger than .50 {Hair, 2019 #1439}, 

		- Discriminant validity  
			Discriminant validity is the extent to which a construct is empirically distinct from other constructs in the structural model {Hair, 2019 #1439@9}.
			
			Fornell-Larcker criterion {Fornell, 1981 #1299}, which is assessed by comparing square roots of AVE and find if they are larger than correlations between constructs {Moody, 2014 #1022}{Hair, 2019 #1439}
			
			or
			
			Heterotrait-monotrait ratio (HTMT) is smaller than .90 (for similar constructs) or .80 (for different constructs) {Hair, 2019 #1439}, 

	- Establishing factorial validities of Formative Constructs  
		- Convergent validity  
			comparing alternative single item measure. correlation should be higher than 0.70.
			
			Often omitted

		- Theoretical support for factorial validity  
			{Diamantopoulos, 2001 #1185}

		- Checking for multicollinearity issues   
			Multicollinearity issues were assessed by observing variance inflation factor (VIF) whcich were within ideal/acceptible range (<3/<3.5) and VIF of >5 is potentially problematic.

		- Indicator weights’ statistical significance and relevance  
			check if the indicator weights are statistically significant. An indicator should be deleted if its weight is insignificant and loadings are below 0.50 and significant.

	- Assessing the structural model  
		- Collinearity assessment  
			Collinearity should be examinde to make sure the regression results were not biased (VIF<5/3).

		- R2  
			a measure of the model’s explanatory power, i.e., in-sample predictive power
			
			.75, .50, .25 is substantial, moderate, weak

		- f2 (optional)  
			reported if editors are reviewers request
			
			report to the presence of partial or full mediation
			
			.02, .15, .35 is small, medium, and large

		- Q2  
			model’s predictive accuracy for an endogenous construct
			
			0, .25, .50 depict small,  medium, and large predictive relevance
			
			use blindfolding from smartPLS

		- Q2predict  
			Q2predict of the key target construct is higher than the most naïve benchmark (>0)

		- PLSpredict  
			comparison of root mean squared error (RMSE) between PLS-SEM analysis and linear regression model (LM) benchmark
			
			in “MV Prediction Summary” in PLS predict of smartPLS
			
			if RMSE of
			
			all indicators of LM<PLS, then no predictive power
			
			LM≤PLS, then low predictive power
			
			LM≥PLS, then medium predictive power
			
			none is LM<PLS, then high predictive power

		- bootstrapping  
			significance and relevance of the path coeficients. mediation analysis is optional
			

		- IPMA (optional)  
			importance-performance map analysis.
			optional
			

		- BIC and GM (optional)  
			comparing models

	- Robustness Check  
		- CTA-PLS (optional)  
			if CI lower and upper are
			-, - then formative
			+, - then formative
			-, + then reflective
			
			this is optional as theoretical reasoning is primary approach

		- Nonlinear effects (optional)  
			a significant test statistic in any of the partial regreassions indicates a potential nonlinear effect

		- Endogeneity (optional)  
			test for endogeneity when a construct is omitted for any reason

		- Unobserved heterogeneity  
			rather a part of data is different than others
			
			1. run FIMIX (finite mixture) segmentation for  1, 2, 3, … n segments
			2. compare the Segment Sizes and see if they are too small; omit segment numberst that are too small
			3. put together the Fit Indicies for each test together in one table
			4. Identify the smallest number from each row:
			AIC: Choose fewer clusterst than this one
			MDL5: Choose more clusters than this one
			AIC3 & CAIC: Choose that number of cluster if they are the same
			AIC4 & BIC: Choose that cluster number if the minimum is in the same column
			EN should be>.5
			
			5. if they are in cluster 1, then there is no segmentation in the data.

<h1>Simpler Description</h1>
with SmartPLS directions

see page 73 of the Hair et al. book for the figure from https://www.smartpls.com/documentation/functionalities/thresholds/

**PLS-SEM algorithm results report including consistent PLS (PLSc)**

--> Final results --> Outer loadings: green if >= 0.7 and red if < 0.7

--> Quality criteria --> f-square: green if >= 0.15 (moderate), black if >= 0.02 (small), and red if < 0.02 (negligible)

--> Quality criteria --> Construct reliability and validity --> Cronbach's alpha: green if >= 0.7 and red if < 0.7

--> Quality criteria --> Construct reliability and validity --> composite reliability rho_a: green if >= 0.7 and red if < 0.7

--> Quality criteria --> Construct reliability and validity --> composite reliability rho_c: green if >= 0.7 and red if < 0.7

--> Quality criteria --> Construct reliability and validity --> average variance extracted (AVE): green if >= 0.5 and red if < 0.5

--> Quality criteria --> Discriminant validity --> heterotrait-monotrait ratio (HTMT): green if <= 0.85, black if <= 0.9, and red > 0.9

--> Quality criteria --> Discriminant validity --> Variance inflation factor (VIF): green if <= 3, black if <= 5, and red if >  5